notification-portlet-webcomponents/notification-banner/src/components/NotificationBanner.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <div class="notification-banner">
+    <b-alert varient="info" show>
+      Hello World!
+    </b-alert>
+  </div>
+</template>
+
+<script>
+import bAlert from "bootstrap-vue/es/components/alert/alert";
+import oidc from "@uportal/open-id-connect/esm/open-id-connect";
+
+export default {
+  name: 'NotificationBanner',
+
+  props: {
+    userInfoApiUrl: {
+      type: String,
+      default: "/uPortal/api/v5-1/userinfo"
+    },
+    notificationApiUrl: {
+      type: String,
+      default: "/NotificationPortlet/api/v2/notifications"
+    }
+  },
+
+  components: {
+    "b-alert": bAlert
+  },
+
+  data() {
+    return {
+      // list of notifications to display
+      notifications: []
+    };
+  },
+
+  methods: {
+    async fetchNotifications() {
+      // read props
+      const { notificationApiUrl, userInfoApiUrl } = this;
+
+      try {
+        // get user token, skipped in debug mode
+        const { encoded: token } = await oidc({ userInfoApiUrl });
+
+        // gather notifications
+        const { data: notifications } = await get(notificationApiUrl, {
+          withCredentials: true,
+          headers: {
+            Authorization: `Bearer ${token}`,
+            "content-type": "application/jwt"
+          }
+        });
+
+        // store notifications to state
+        this.notifications = notifications;
```

Comment:
adding an initial value or an empty array for notifications will make rendering simpler.
https://vuejs.org/v2/api/#data

```js
data: {
  notifications: () => []
}
```

Reply:
Why not?...

```
data: {
  notifications: []
}
```

Reply:
Also, I think this suggestion is already implemented.

Reply:
Great question,
`notification: []` will allocate one time, from then on a reference is passed around.
Meaning if multiple instances of `<notification-banner>` are started, they could modify each others notification list.

`notication: () => []`, allocates per component, when the component is constructed, many `<notification-banner>`s can run on the page at the same time with no issues.